### ISSUE TITLE
add peer review info to deposit

### ISSUE THREAD

**Issue by Melissa37** [elifesciences/elife-crossref-feed#123]
Tasks
- [x] Test adding peer review materials to a Crossref deposit along with the article deposit itself to check whether that is valid and allowed - Answer: No, we cannot mix journal and peer_review deposits in a single deposit file
- [x] Note: peer_review deposits is compatible with Crossref schema version 4.4.1
- [x] Parse more extensive data from `<sub-article>` tags in the `elife-tools` project
- [x] Expand the `elife-article` project objects to parse and store the peer review material details as an attribute of an `Article`
- [x] Programming logic add to `elife-crossref-xml-generation` project to generate peer review Crossref deposit batch files
- [x] Add license to the Peer review deposit
- [x] Populate the author response authors of the peer review deposit (logic already works, we may do this in a workflow, using the parent article authors, in eLife's use case)
- [x] Set the review data value for each peer review (TBD: where to get this data)
- [x] ~~Enhance or replace the editors / reviewers / authors of the decision letter peer review deposit (TBD: figure out from where to get the data)~~ not going to enhance the data outside of what is in the XML already
- [x] Reach QC approval from Production and Editorial on generated samples
- [x] Bot activity
- [x] Bot workflow
- [x] Bot starter
- [x] ~~**Blocked** by~~ adding Google credentials to `elife-bot` https://github.com/elifesciences/issues/issues/5178
- [x] Can start by depositing a few articles manually
- [x] Deposit all peer reviews for all historic articles (tracking in issue https://github.com/elifesciences/issues/issues/5193)
- [x] ~~Add to bot `ScheduleDownstream` to automatically add to the outbox upon publication~~ Adding as a pre-publication activity, but will only deposit after the research article DOI exists and some other stipulations
- [x] Add to bot `cron.py` to run this peer review deposit workflow regularly

Optional:
- [x] May want to ~~/ need to~~ upgrade to use Crossref schema version 4.4.2 (eLife is currently using 4.4.1) (testing I did around March 2019 showed we do not need to upgrade to 4.4.2 for the peer review deposit to be valid) - Note: __Skipping this for now__

**Comment by Melissa37** [comment]
Reference: https://elifesciences.atlassian.net/browse/ELPP-3189

**Comment by Melissa37** [comment]
@gnott Can I add this to our board as we can now proceed with it now the relevant info is in the JATS XML?
Let's talk about it at our next catch up!

**Comment by Melissa37** [comment]
Example from Crossref schema, updated with eLife content

```
<body>
<peer_review stage="pre-publication" type="editor-report">
<contributors>
<person_name contributor_role="editor" sequence="first">
<given_name>Rutz</given_name>
<surname>Christian</surname>
</person_name>
</contributors>
<titles>
<title>Decision letter: Immune mediated hookworm clearance and survival of a marine mammal decreases with warmer ocean temperatures</title>
</titles>
<review_publication_date>
<year>2018</year>
</review_publication_date>
<related_item>
<inter_work_relation relationship-type="isReviewOf" identifier-type="doi">10.7554/eLife.38432</inter_work_relation>
</related_item>
</program>
<doi_data>
<doi>10.7554/eLife.38432.034</doi>
<resource>https://elifesciences.org/articles/38432#SA1</resource>
</doi_data>
</peer_review>
<peer_review stage="pre-publication" type="author-comment">
<contributors>
<person_name contributor_role="author" sequence="first">
<given_name>Mauricio</given_name>
<surname>Seguel</surname>
</person_name>
<person_name contributor_role="author" sequence="second">
<given_name>Felipe</given_name>
<surname>Montalva</surname>
</person_name>
<person_name contributor_role="author" sequence="third">
<given_name>Diego</given_name>
<surname>Perez-Venegas</surname>
</person_name>
...

            </contributors>
            <titles>
                <title>Decision letter: Immune mediated hookworm clearance and survival of a marine mammal decreases with warmer ocean temperatures</title>
            </titles>
            <review_publication_date>
                <year>2018</year>
            </review_publication_date>
                <related_item>
                    <inter_work_relation relationship-type="isReviewOf" identifier-type="doi"
                        >10.7554/eLife.38432</inter_work_relation>
                </related_item>
            </program>
            <doi_data>
                <doi>10.7554/eLife.38432.035</doi>
                <resource>https://elifesciences.org/articles/38432#SA2</resource>
            </doi_data>
        </peer_review>
    </body>
```

**Comment by gnott** [comment]
You can probably explain further. My guess so far is in the main article deposit to omit the decision letter and author response components, deposit those separately, but included them as a `<related_item>` as an `<inter_work_relation relationship-type="isReviewOf">`. Is it so you can add reviewer names to the sub article?

**Comment by Melissa37** [comment]
>Is it so you can add reviewer names to the sub article?

No, it's a Crossref initiative for publishers to submit peer review materials with their own DOIs:

https://support.crossref.org/hc/en-us/articles/115005255706-Peer-Reviews#rel

Now that we publish reviewer names and senior and reviewing editor names, I need to double check with @acollings what should be deposited with Crossref.

We can chat more on our next call.

**Comment by Melissa37** [comment]
Crossref fixed issues.
Sample:

```<?xml version="1.0" encoding="UTF-8"?>
<doi_batch version="4.4.2" xmlns="http://www.crossref.org/schema/4.4.2"
    xmlns:ai="http://www.crossref.org/AccessIndicators.xsd"
    xmlns:ct="http://www.crossref.org/clinicaltrials.xsd"
    xmlns:fr="http://www.crossref.org/fundref.xsd" xmlns:jats="http://www.ncbi.nlm.nih.gov/JATS1"
    xmlns:mml="http://www.w3.org/1998/Math/MathML" xmlns:rel="http://www.crossref.org/relations.xsd"
    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
    xsi:schemaLocation="http://www.crossref.org/schema/4.4.2 http://www.crossref.org/schemas/crossref4.4.2.xsd">
    <head>
        <!-- Will have to decide how to name the doi_batch_id -->
        <doi_batch_id>elife-crossref-38432-relations</doi_batch_id>
        <timestamp>20181120160009</timestamp>
        <depositor>
            <depositor_name>eLife</depositor_name>
            <email_address>production@elifesciences.org</email_address>
        </depositor>
        <registrant>eLife</registrant>
    </head>
    <body>
    <peer_review stage="pre-publication" type="editor-report">
        <contributors>
            <person_name contributor_role="editor" sequence="first">
                <given_name>Rutz</given_name>
                <surname>Christian</surname>
            </person_name>
        </contributors>
        <titles>
            <title>Decision letter: Immune mediated hookworm clearance and survival of a marine mammal decreases with warmer ocean temperatures</title>
        </titles>
        <review_date>
            <month>01</month>
            <day>02</day>
            <year>2018</year>
        </review_date>
        <program xmlns="http://www.crossref.org/relations.xsd">
        <related_item>
            <inter_work_relation relationship-type="isReviewOf" identifier-type="doi">10.7554/eLife.38432</inter_work_relation>
        </related_item>
        </program>
        <doi_data>
            <doi>10.7554/eLife.38432.034</doi>
            <resource>https://elifesciences.org/articles/38432#SA1</resource>
        </doi_data>
    </peer_review>
        <peer_review stage="pre-publication" type="author-comment">
            <contributors>
                <person_name contributor_role="author" sequence="first">
                    <given_name>Melissa</given_name>
                    <surname>Harrison</surname>
                </person_name>
            </contributors>
            <titles>
                <title>Author response: Immune mediated hookworm clearance and survival of a marine mammal decreases with warmer ocean temperatures</title>
            </titles>
            <review_date>
                <month>01</month>
                <day>02</day>
                <year>2018</year>
            </review_date>
            <program xmlns="http://www.crossref.org/relations.xsd">
                <related_item>
                    <inter_work_relation relationship-type="isReviewOf" identifier-type="doi">10.7554/eLife.38432</inter_work_relation>
                </related_item>
            </program>
            <doi_data>
                <doi>10.7554/eLife.38432.035</doi>
                <resource>https://elifesciences.org/articles/38432#SA1</resource>
            </doi_data>
        </peer_review>
    </body>
</doi_batch>  
```

**Comment by Melissa37** [comment]
This is connected with https://github.com/elifesciences/elife-crossref-feed/issues/140 and https://github.com/elifesciences/XML-mapping/issues/87

The DOIs will have a structure of:
10.7554/eLife.42687.DL
10.7554/eLife.42687.AR

**Comment by gnott** [comment]
Reviewing this feature again, what might work is to add the `<peer_review>` parts to the existing Crossref batch file eLife produces for each journal article. That is if this review content will still be part of the one article JATS file. 

If you want to supply the review content as separate files, then it might turn out slightly differently. You can please clarify the input format(s) we need to support in the Deposit Crossref workflow.

**Comment by gnott** [comment]
I tried validating a deposit that has both `<journal>` and `<peer_review>` elements in the `<body>` and it is not valid. The documentation on this is clear on for example http://data.crossref.org/schemas/crossref4.4.1.xsd, it says `"It is not possible to mix content types within a single registration file."`. So peer review deposits will most certainly have to be a separate deposit file.

**Comment by gnott** [comment]
@Melissa37 - do you suppose eLife can always set `peer_review stage="post-publication"` on the `<peer_review>` tag, or should we also be setting `peer_review stage="pre-publication"` as well when appropriate?

You have `"pre-publication"` in your latest sample above so I wasn't sure.

**Comment by Melissa37** [comment]
HI G it should always be ````peer_review stage="pre-publication'````

The peer review is happening before publication.

Can you confirm this is always set to ````peer_review stage="pre-publication'````
Otherwise it's wrong!

**Comment by gnott** [comment]
Ah, thanks - I misinterpreted its meaning as relating to the publication status of the article itself. You show it relates to at what stage the article was at when the review is done 👍

**Comment by gnott** [comment]
I completed a first valid demonstration of generating peer review deposits. 

In your example above, I noticed one tiny detail which is hardly worth mentioning: the author response `<resource>` would probably be `38432#SA2` and not `38432#SA1`, because '#SA1` is already used by the decision letter, unless both documents would be located at the same place.

I think in the end the peer review documents may be at DOI and and URL that is different from the parent article, so what it is in the example doesn't matter. I do need to work on some logic of how to parse or set these addresses for eLife examples.

Next up is to add values to the `elife-article` project to support review articles. I was able to reuse almost everything from an `Article` object. Parsing the values from an XML file will be the bulk of the coding that remains.

**Comment by Melissa37** [comment]
>In your example above, I noticed one tiny detail which is hardly worth mentioning: the author response <resource> would probably be 38432#SA2 and not 38432#SA1, because '#SA1` is already used by the decision letter, unless both documents would be located at the same place.

Thanks G, correct! Sorry for the error. I'll correct it

**Comment by Melissa37** [comment]
Hi @gnott been re-reviewing this in the light of the JATS4R work and what else I know since I opened this ticket :-)  I've also looped in people we need input from in the specific themes below...

**ORCIDs**
- It would be really good if we can add the ORCIDs for the people. I assume this would involve interrogating the EJP metadata to pull it from there? I am adding @JGilbert-eLife to this ticket as he might be able to shed some light as to how ORCIDs are stored in the database and where/how they are available. I suspect this could get complex so it might be worth us discussing it together on the next call, with James. Feedback from ORCID and Crossref is that it would be preferable to add the ORCIDs to the deposit if we have them. To begin with this can bypass production, as necessary (ie take direct from EJP output), but ultimately it would be good to display these ORCIDs in the production process and on the website.

**License data**
Can we add this too? As we're so good at providing our license data in the rest of our deposits I'd like to make this open and accessible too.

**Recommendation**
I'm looping in @acollings here for this one.
The options are:

- major-revision
- minor-revision
- reject
- reject-with-resubmit
- accept

It would be nice to be able to submit this data too, but right now it might be too difficult as EJP might not be programmatically storing this, I am thinking of Research Communication, We cannot assume every accepted article should be "accept" can we Andy?

 **Future**
We are doing a lot of work around peer review publication and there is likely to be change in the future. While we're testing editorially and doing experiments, we're going for the lowest common denominators that require no development work in order to do the editorial experiments!

Cheers
M

**Comment by Melissa37** [comment]
**Date**
@acollings another issue:

The date we're adding into this Crossref deposit.
We are meant to add the date of the work, not date of publication. At a minimum, we can add the year and not specific day and month details.
Do you have an opinion on this, because again we could use the EJP data to give us this information....again looping in @JGilbert-eLife !

**Comment by JGilbert-eLife** [comment]
The ORCID information is present in the XML dumps from eJP, so I assume it could be accessed via the data pipeline?

<person>
      . . . .
      <first-name>Moses</first-name>
      <institution>New York University Langone Medical Center</institution>
      <last-name>Chao</last-name>
      <login>mchao</login>
      <memberships>
        <membership>
          <member-id>0000-0002-6969-3744</member-id>
          <member-type>ORCID</member-type>
        </membership>
      </memberships>
      <middle-name>V</middle-name>
      . . . .
    </person>

(I claim no responsibility for the insane order this information appears in in the dump)

For dates, we have a selection but I suppose the date the review is submitted to us counts as the point at which it is . . . completed as a piece of work? It's the point it's 'published to the editor and reviewers, that might make most sense to use? We should be able to access that information through the dumps as well, I think.
```

**Comment by acollings** [comment]
Hi @Melissa37 Thanks for looping me in.

On the issue of the **Recommendation**: there are a couple of reasons why I suggest that we avoid assigning a recommendation. First, what we're publishing is essentially the revise letter, asking for revisions, although we are publishing it along with the accepted paper. We don't differentiate between minor and major revisions: that distinction is pretty arbitrary. (We also have complications with Research Communications as you note.) Second, the direction of travel (long term) is to move away from the notion of accept and reject decisions (i.e., publish first), so it would be nice to avoid reinforcing them in this context. Is that okay?

On the issue of the **Date**: the date of the work in our case corresponds with the date of the decision after peer review (if we keep things simple). If it's easy enough to extract the decision date (of the full submission) from eJP, I'd suggest doing so. If we only list the year, we'll have cases where the decision was rendered in 2019 but the paper was published in 2020, so that won't always be right.

Many thanks, Andy

**Comment by gnott** [comment]
I refreshed my memory of where I left this feature last time, and only the output is in draft form.

The output itself is (I think) and is planned to be based on a regular `Article` object, which has authors, license, dates, etc.

The `<sub-article>` in the JATS file can form the basis of the data, but if things like the ORCID are not specified there, then the `Article` object can be added to with data from external sources. These would include editor ORCID values (not present in the XML), alternate dates.

For the authors of an author response, I suppose those would be all the authors of the article itself, or would it be only the corresponding author(s)? Those may or may not have ORCID values in the XML already for inclusion.

**Comment by Melissa37** [comment]
>If we only list the year, we'll have cases where the decision was rendered in 2019 but the paper was published in 2020, so that won't always be right.

@acollings can you clarify why that won't be right, James and I were confused about this. 

Also, I assume the date for each of these assets should be different from each other as they are written at different time points in the process? ie the decision letter is written and then the author responds to it. 

Can you clarify/confirm which dates from EJP we should publish for each of these documents?

Thanks!

**Comment by Melissa37** [comment]
>The output itself is (I think) and is planned to be based on a regular Article object, which has authors, license, dates, etc.

However, the dates and authors are different to the dates of the main XML document

>The <sub-article> in the JATS file can form the basis of the data, but if things like the ORCID are not specified there, then the Article object can be added to with data from external sources. These would include editor ORCID values (not present in the XML), alternate dates.

This is interesting and good. Perhaps we could also specify other data should be taken from extenral sources too, such a date and authors...

>For the authors of an author response, I suppose those would be all the authors of the article itself, or would it be only the corresponding author(s)? Those may or may not have ORCID values in the XML already for inclusion.

Yes, all authors. I don't think we need to try and pull the ORCIDs for this asses from EJP output - what we have in the final JATS XML is good

Thanks!

Also, I have updated the ticket in Exeter's repo with a top level list of issues and questions that cross-reference a lot of this. I suggest we wait for them to digest and in a week or two have a joint call with them.

**Comment by gnott** [comment]
Do we intend to deposit peer reviews for all previous articles? If so, will the component DOI values on those in older articles remain the same?

**Comment by Melissa37** [comment]
I have now emailed Crossref for their input on the decision/question.

**Comment by Melissa37** [comment]
>For dates, we have a selection but I suppose the date the review is submitted to us counts as the point at which it is . . . completed as a piece of work? It's the point it's 'published to the editor and reviewers, that might make most sense to use? We should be able to access that information through the dumps as well, I think.

@JGilbert-eLife Andy said the revised decision date is the one we want - just using the Year value

**Comment by Melissa37** [comment]
@JGilbert-eLife Andy says date for author response is revised qc approved - only use year value

**Comment by Melissa37** [comment]
Updated with decisions from the call yesterday: @acollings , Melissa/ @gnott :

**Decision letter**

- Will use "Revise Decision" date
@JGilbert-eLife to tell @gnott where this is in the EJP export CSV files and also help decipher where it would be in data hub

- Will use the Editor and Reviewer names provided in the final XML document
eLife Editors (different titles/roles) will be converted to simple editor in Crossref feed, and reviewers to reviewers

- Will use ID of paper + names of people from XML to query for ORCIDs from EJP metadata. This will prevent accidental pull of ORCIDs for someone else with the same name and also accidental publication of reviewers who don't want to be named.

There is 70% chance editors have an ORCID in EJP and about a third of reviewers have ORCIDs (but remember a third of reviewers reveal their name)

**Author response**

- Will use "Revised QC approval" date
@JGilbert-eLife to tell @gnott where this is in the EJP export CSV files and also help decipher where it would be in data hub

- Will add all author names (not just the corresponding author) and will add all available ORCIDs from the XML. No need to query EJP for ORCIDs
In Crossref schema there is a First author and then other authors

**Archive**
All necessary dates are there in the archive.
Older articles will have fewer and fewer names of editors/reviewers as we go back in time. Do not collect any names from EJP, just work with what we have in the XML.
OK to add ORCIDs for authors in the archive if their ORCIDs are not in the published papers.

**Testing**
Graham to create test samples from Production and Andy to QC before sending anything live to Crossref

**Comment by Melissa37** [comment]
@JGilbert-eLife to check re appropriate dates and dates Daniel has in data pipeline - communicate with @gnott which this will be called.

Translation job - required James!!

**Comment by gnott** [comment]
Thanks for the detailed examples @JGilbert-eLife! I removed the blocker label on this issue, and next steps include: 

- [ ] me to start building a bot activity for running a workflow, and 
- [ ] Michael will help me to connect the bot to the data

**Comment by gnott** [comment]
I forgot this issue is in the Crossref project repo, and I added some additional checkboxes to the first comment regarding implementation and deployment in the `elife-bot` for eLife specifically. It seemed most straightforward to just use this one ticket to track the entire process for this feature.

**Comment by gnott** [comment]
Is becoming unblocked, I will remove the blocker tag as credentials are now in process of being integrated.

**Comment by Melissa37** [comment]
This is not assigned to anyone, can we assign it to you now @gnott ?

**Comment by gnott** [comment]
I've self-assigned it. It is in progress right now, related to another ticket. Next step is setting things up for automated delivery.

**Comment by gnott** [comment]
I would like to postpone upgrading the schema we use to `4.4.2` to a later date, since it is not required for depositing peer reviews. That makes all the todo items on this issue now finished.
